### PR TITLE
Fix slack bot formatting and add a channel whitelist for sig mention reiteration.

### PR DESCRIPTION
**Pull request by cjwagner**
/cc @rmmh @BenTheElder

### CODE REVIEW

File: prow/plugins.yaml [hunk 1]
```diff
@@ -27,6 +27,37 @@ label:
...
+  - sig-openstack
+  - sig-release
+  - sig-rktnetes
+  - sig-scale
+  - sig-scheduling
+  - sig-service-catalog
+  - sig-storage
+  - sig-testing
+  - sig-ui
+  - sig-windows
```
Comment (BenTheElder) [inline prow/plugins.yaml]: is this just all the sigs?

Comment (cjwagner) [reply prow/plugins.yaml]: Its now all the sigs except sig-auth.

**Comment by kargakis** [comment]
Why is this part of the label plugin as opposed to the slackevents plugin?

**Comment by cjwagner** [comment]
> Why is this part of the label plugin as opposed to the slackevents plugin?

Its part of label because the label plugin already matches on the same sig regex. Slackevents was created to react to push events, i think new slack behavior would belong there if it didn't fit into a different plugin already (like label).

That being said, the label plugin is becoming a bit unwieldy. Some parts like the milestonemaintainer status labels should have been put into a separate plugin.

I want to refactor some of the plugins to register a new `CommentHandler` instead of having to register a handler for every type of event that can include a text body. That will make it easier to split up the label plugin since the boiler plate related to registering for all the types of events will be removed.

**Comment by kargakis** [comment]
> Its part of label because the label plugin already matches on the same sig regex. Slackevents was created to react to push events, i think new slack behavior would belong there if it didn't fit into a different plugin already (like label).

The label plugin should really handle only labeling and the slackevents plugin should handle all interactions with slack. Bleeding logic between plugins is not going to end up being fun.

**Comment by BenTheElder** [comment]
I agree with @kargakis. I'm not sure how we should solve sharing matching generally across plugins but I think we should probably keep all of the slack logic in the slack plugin.

**Comment by fejta** [comment]
Seems like we're conflating three things into a single plugin:
1) Add a sig-foo label on a `@kubernetes/sig-foo-whatever` text
2) Echo the message back to the comment (since `@kubernetes/sig-foo-whatever` will not notify someone if the comment author isn't in the k8s org)
3) Echo the message to slack

Agree it sounds like a good idea to decouple at least (1) from (2)-(3)?

File: prow/plugins/label/label_test.go [hunk 2]
```diff
@@ -584,11 +585,11 @@ func TestLabel(t *testing.T) {
 
 		for i := 0; i < len(fakeRepoFunctions); i++ {
 			fakeClient, ae := fakeRepoFunctions[i](tc.body, tc.commenter, tc.repoLabels, tc.issueLabels)
-			fakeMilestoneId := 123456
+			fakeConfig := plugins.Label{MilestoneMaintainersID: 123456}
```
Comment (rmmh) [inline prow/plugins/label/label_test.go]: maybe 'fakeLabel' instead

File: prow/plugins.yaml [hunk 3]
```diff
@@ -27,6 +27,36 @@ label:
   # responsible for maintaining the milestones:
   # curl -H "Authorization: token <token>" https://api.github.com/orgs/<org-name>/teams
   milestone_maintainers_id: 2460384
+  slacksigs:
```
Comment (rmmh) [inline prow/plugins.yaml]: Maybe `slackchannels` instead? More importantly, what does "label" even mean here? Please document the yaml.

File: prow/plugins/label/label.go [hunk 4]
```diff
@@ -287,21 +287,31 @@ func handle(gc githubClient, log *logrus.Entry, ae assignEvent, sc slackClient,
...
+			log.WithError(err).Errorf("Could not create comment \"%s\".", msg)
 		}
+	}
 
-		// If sig matches then send a notification on slack.
-		for _, sig := range sigMatches {
-			msg := fmt.Sprintf("Message: ```%s```\nIssue: %d, %s\nUrl: %s.", ae.body, ae.issue.Number, ae.issue.Title, ae.issue.HTMLURL)
-			if err := sc.WriteMessage(plugins.FormatResponseRaw(ae.body, ae.url, ae.login, msg), "sig-"+sig[1]); err != nil {
-				log.WithError(err).Error("Failed to send message on slack channel: ", "sig-"+sig[1], " with message ", msg)
+	for _, match := range sigMatches {
```
Comment (rmmh) [inline prow/plugins/label/label.go]: We're going to let non-members ping slack?

File: prow/plugins.yaml [hunk 5]
```diff
@@ -27,6 +27,36 @@ label:
...
+  - sig-api-machinery
```
Comment (kargakis) [inline prow/plugins.yaml]: Btw, this list should include only sig-testing for start. Make an announcement in kubernetes-dev and any interested SIGs should be able to open PRs and add themselves.

Comment (fejta) [reply prow/plugins.yaml]: I don't think that is how people operate...

There are huge outcome differences depending on whether you ask new employees to opt-in or opt-out of retirement planning like a 401K. 

Maybe limit it to sig-testing until we have more confidence the plugin is good and then foist it upon everyone else?

File: prow/plugins/label/label.go (hunk 4 above)
Comment (cjwagner) [reply prow/plugins/label/label.go]: We already reiterate mentions from non-members on github so I figured we would want to allow non-members to ping slack as well. I could easily change this to only message slack for comments left by org members if that is preferred though.

**Comment by cjwagner** [comment]
Refactored. PTAL.

I'll split up the rest of the label plugin and switch plugins to use the generic comment events in a follow up PR.

**Comment by BenTheElder** [comment]
/lgtm
:+1: to the follow up plan.

**Pull request merged**